### ISSUE TITLE
AI does not reliably upgrade/morph buildings

### ISSUE THREAD

**Issue by andy5995** [ZetaGlest/zetaglest-source#119]
Buildings can be morphed, but the AI doesn't do it in ever case. Right now, there really aren't many morph-able buildings in ZG (but the feature does exist)

@MirceaKitsune wrote
>So one thing I should ask and / or note about the AI: One huge issue I ran into with factions I was developing for MegaGlest years ago, was that the AI will never upgrade buildings. This is because apparently, it does not scan the advantages of what an upgraded unit can produce, it only looks for attacks. Does the AI in Zetaglest know to upgrade buildings, when morphing to a new one offers advantages such as new units being able to produce?
In one faction, I had to give the upgradable building an attack just to trick the AI into upgrading to it so the whole faction would be playable. It fits well in that case so I'm leaving it as is... however for some factions and units that approach isn't possible.

With the Indian faction, I once made a test mod in which the main teepee could morph into a *flying* main teepee. The AI did morph it (iirc, I think it would morph it every time).

More recently, however, I observed that the Elves never upgrade the corral into a corral_gryphon_nest. I'm not sure why yet. The only other upgradeable buildings I can think of off the top of my head are the Greek barracks. I haven't yet checked to see if the AI ever upgrades them.

Thank you for bringing up this issue.

**Comment by andy5995** [comment]
>the Flying Teepee likely has an attack so it will account for it.

That's true! I gave it an attack capability (like a beehive). Ok, that makes it a lot more clear.

So one temporary work-around might be to give a building attack capability.. require EP for the attack, but don't give it any EP, or a skill to regenerate EP.

**Comment by KeithSammut** [comment]
oh this issue came up with the flying container in the specialists
